lib/mongo_oplog_backup/restore.rb

```diff
@@ -104,14 +124,15 @@ def validate_continuity!(source_files)
     end
 
     def create_temp_oplog_dir dir, filename
-      temp_dir = File.join(dir, 'tmp-restore', filename)
+      temp_dir = File.join(dir, 'tmp-restore', File.basename(filename))
       temp_file_path = File.join(temp_dir, 'oplog.bson')
       FileUtils.mkdir_p(temp_dir)
       File.link(filename, temp_file_path)
       return temp_file_path
     rescue Errno::EEXIST => e
       # Probably, 'new_name' already exists when creating link.
-      MongoOplogBackup.log.warn("Cannot create temporary oplog.bson link: #{e.message}")
+      MongoOplogBackup.log.warn("Temporary oplog.bson link already exists: #{e.message}. Assuming valid.")
```

Comment:
Sorry, missed the fact that you are creating a separate temp dir per file, not re-using the same one.